modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/EagerTtlTest.java

```diff
@@ -0,0 +1,201 @@
+import org.apache.ignite.configuration.DataStorageConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.IgniteInternalFuture;
+import org.apache.ignite.internal.processors.affinity.AffinityTopologyVersion;
+import org.apache.ignite.internal.processors.cache.distributed.dht.preloader.GridDhtPartitionsExchangeFuture;
+import org.apache.ignite.internal.processors.cache.distributed.dht.preloader.PartitionsExchangeAware;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.testframework.ListeningTestLogger;
+import org.apache.ignite.testframework.LogListener;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.apache.ignite.transactions.Transaction;
+import org.apache.ignite.transactions.TransactionConcurrency;
+import org.apache.ignite.transactions.TransactionIsolation;
+import org.junit.Test;
+
+/**
+ * Test checks a work of the TTL policy.
+ */
+public class EagerTtlTest extends GridCommonAbstractTest {
+
+    /** Text will print to log when assertion error happens. */
+    private static final String ASSERTION_ERR = "java.lang.AssertionError: Invalid topology version [topVer=" +
+        AffinityTopologyVersion.NONE +
+        ", group=" + DEFAULT_CACHE_NAME + ']';
+
+    /** Text will appear when an assertion error happens. */
+    public static final String ANY_ASSERTION_ERR = "java.lang.AssertionError";
+
+    /** Expiration time. */
+    public static final int EXPIRATION_TIME = 1_000;
+
+    /** Count of entries. */
+    public static final int ENTRIES = 100;
+
+    /** Cache eager ttl falg. */
+    private boolean eagerTtl = false;
+
+    /** Listening logger. */
+    private ListeningTestLogger listeningLog;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super.getConfiguration(igniteInstanceName)
+            .setGridLogger(listeningLog)
+            .setClusterStateOnStart(ClusterState.INACTIVE)
+            .setCacheConfiguration(getCacheConfiguration(DEFAULT_CACHE_NAME))
+            .setDataStorageConfiguration(new DataStorageConfiguration()
+                .setDefaultDataRegionConfiguration(new DataRegionConfiguration()
+                    .setMaxSize(200L * 1024 * 1024)
+                    .setPersistenceEnabled(true)));
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        super.beforeTest();
+
+        stopAllGrids();
+
+        cleanPersistenceDir();
+
+        listeningLog = new ListeningTestLogger(log);
+    }
+
+    /**
+     * Creates a configuration for cache with the name specified.
+     *
+     * @param cacheName Cahce name.
+     * @return Cache configuration.
+     */
+    private CacheConfiguration getCacheConfiguration(String cacheName) {
```

Comment:
Could you please add `<Integer, Integer>` (here and below) in order to eliminate compiler warnings?

Reply:
I do not like to do it, because this makes code more verbose.
But it is your wish.

Reply:
On the other hand, this allows us to avoid compiler warnings, which is good I think.